pxf/pxf-json/src/test/java/org/apache/hawq/pxf/plugins/json/JsonExtensionTest.java

```diff
@@ -0,0 +1,288 @@
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add(",,,,,,"); // Expected: malformed json records are transformed into empty rows
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-broken.json"), output);
+	}
+
+	@Test
+	public void testSmallTweets() throws Exception {
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547123646465,text2,patronusdeadly,,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+		output.add("Fri Jun 07 22:45:03 +0000 2013,343136551322136576,text4,SevenStonesBuoy,,-6.1,50.103");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-small.json"), output);
+	}
+
+	@Test
+	public void testTweetsWithNull() throws Exception {
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,,text2,patronusdeadly,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/null-tweets.json"), output);
+	}
+
+	@Test
+	public void testSmallTweetsWithDelete() throws Exception {
+
+		List<String> output = new ArrayList<String>();
+
+		output.add(",,,,,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547123646465,text2,patronusdeadly,,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-small-with-delete.json"), output);
+	}
+
+	@Test
+	public void testWellFormedJson() throws Exception {
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "created_at"));
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547123646465,text2,patronusdeadly,,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets-pp.json"), output);
+	}
+
+	@Test
+	public void testWellFormedJsonWithDelete() throws Exception {
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "created_at"));
+
+		List<String> output = new ArrayList<String>();
+
+		// output.add(",,,,,,");
```

Comment:
I am not sure what was the purpose of the deleted json records in the original pxf-json code. Perhaps @adamjshook would remember? 
But i've kept the test because it shows that the identifier will filter out the records not containing the identifier (nothing to do with deletion though).